src/Components/Import/Import.tsx

```diff
@@ -1,32 +1,178 @@
+          onError={() => alert(file + " is unable to play")}
+          id={file.name}
+          onReady={addMedia}
+        />
+      );
+      let newMedia = new Media(file.name, "video", el);
+      addFiles.push(newMedia);
+    } else {
+      showError("invalid file " + file.name);
+    }
+  };
+  const MAXLEN = 30;
 
+  const dropHandler = (e: React.DragEvent<HTMLDivElement>) => {
+    setOnDropState(true);
+    e.preventDefault();
+    const dup: Media[] = [];
+
+    if (e.dataTransfer.items) {
+      // Use DataTransferItemList interface to access the file(s)
+      for (let i = 0; i < e.dataTransfer.items.length; i++) {
+        // If dropped items aren't files, reject them
+        if (e.dataTransfer.items[i].kind === "file") {
+          let file = e.dataTransfer.items[i].getAsFile();
+          if (file == null) {
+            return;
+          }
+          setOnDropState(true);
+          mediaElement(dup, file);
+        }
+        setOnDropState(true);
+      }
+    } else {
+      // Use DataTransfer interface to access the file(s)
+      for (let i = 0; i < e.dataTransfer.files.length; i++) {
+        mediaElement(dup, e.dataTransfer.files[i]);
+      }
+    }
+    setOnDropState(true);
+    setMedias(dup);
+  };
+  const dragOverHandler = (e: React.MouseEvent<HTMLDivElement>) => {
+    e.preventDefault();
+  };
+  const dragEnterHandler = (e: React.MouseEvent<HTMLDivElement>) => {
+    e.preventDefault();
+    if (!onDragState) {
+      setOnDragState(true);
+    }
+  };
 
-export const Import: React.FC<Props> = ({}) => {
-// function Import() {
-  const [count, setCount] = useState<TextNode>({text: 'ughh'});
   return (
-    <div>
-      quiz!
-      <form action="/action_page.php">
-        <input type="file"></input>
-        <input type="submit" value="Submit"></input>
-      </form>
+    <div
+      className={styles.canvaHomePage}
+      onDragEnter={dragEnterHandler}
+      onDrop={dropHandler}
+      onDragOver={dragOverHandler}
+    >
+      <ErrorModal/>
+      {!onDropState && onDragState ? (
+        <div
+          className={styles.dropModal}
+          onDragLeave={(e) => {
+            setOnDragState(false);
+            e.preventDefault();
+          }}
+        >
+          <div className={styles.dotted}>
+            <img className={styles.characterIcon} src={character} alt="drag file here" />
+            <div className={styles.dropModalText}>Drop Your File Here</div>
+          </div>
+        </div>
+      ) : null}
+      {onDropState && onDragState ? (
+        <div
+          className={styles.dropModal}
+          onDragLeave={(e) => {
+            setOnDragState(false);
+            e.preventDefault();
+          }}
+        >
+          <div className={styles.dotted}>
+            {/* filename (key) to JSX element (value) mapping */}
+
+            {medias.map((media: Media, index: number) => {
+              return (
+                <div
+                  key={media["filename"]}
```

Comment:
nit: media.filename
only need to use square bracket notation (`media["filename"]`) if the property value is variable